src/client.rs

```diff
@@ -84,6 +84,84 @@ pub struct Connection<T: Read + Write> {
     pub greeting_read: bool,
 }
 
+/// A builder for the append command
+pub struct AppendCmd<'a, T: Read + Write> {
```

Comment:
`AppendCmd` probably shouldn't be public.

Reply:
Hmm, why not? I think it probably needs to be.

Reply:
Also, I think this struct should have `#[must_use]` added to it.

Reply:
My argument for not having it public was that I wanted to hide it "behind" the `append()` function so as to not let the user manually construct it.

But you are right it needs to be pub when used as a return type of pub function.